### ISSUE TITLE
bug : shows that they queries are same, when they are not.

### ISSUE THREAD

**Issue by argahsuknesib** [argahsuknesib/rspql-query-equivalence#2]
For query one with the basic graph pattern,
```
    ?s rdf:type :Person .
    ?s rdf:type :Student .
```
and for query two,
```
?s rdf:type :Person .
?p rdf:type :Student .
```
The library shows that the queries are the same, which they are not. 
As in the query one, there is a join between the :Person and :Student, whereas in the second query it is not the case.

**Comment by argahsuknesib** [comment]
A pre-processing step, where you take in account the variable when converting it to a blank node can be helpful, or skolemisation maybe? Further investigation is required in this direction.

**Comment by argahsuknesib** [comment]
Turns out, there was a fault in the blank node creation. Fixed with b89aa2f3b76ad491c28b2190d2421f64761d4104
